pandas/core/internals/blocks.py

```diff
@@ -1864,6 +1864,11 @@ def where(
 
         return [self.make_block_same_class(result, placement=self.mgr_locs)]
 
+    def equals(self, other) -> bool:
+        if self.dtype != other.dtype or self.shape != other.shape:
```

Comment:
why do you need this check here? (its already done on the .values no?)

Reply:
Hmm, good question. I just copied that from the base Block.equals method (so it's done there as well). There, `array_equivalent` is used. Maybe that is less strict and the extra check is needed, which is not needed here.

Reply:
At least from the docstring of `array_equivalent` (not sure how up to date that is), using that method indeed requires a more strict check in advance of calling it:

https://github.com/pandas-dev/pandas/blob/3ed7dff48bb4e8c7c0129283ff51eccea3a0f861/pandas/core/dtypes/missing.py#L360-L363

But so, for EA.equals that is not needed, so will remove that check here.